src/vm.c

```diff
@@ -248,6 +275,46 @@ int vm_irq_line(vm_t *v, int irq, int level)
     return 0;
 }
 
+int vm_irq_trigger(vm_t *v, int irq)
+{
+    if (vm_irq_line(v, irq, 1) < 0)
+        return -1;
+    if (vm_irq_line(v, irq, 0) < 0)
+        return -1;
+    return 0;
+}
+
+void *vm_host_to_guest(vm_t *v, void *host)
+{
+    return host - (uintptr_t) v->mem;
+}
+
+void *vm_guest_to_host(vm_t *v, void *guest)
+{
+    return (uintptr_t) v->mem + guest;
+}
+
+void vm_irqfd_register(vm_t *v, int fd, int gsi, int flags)
+{
+    struct kvm_irqfd irqfd = {.fd = fd, .gsi = gsi, .flags = flags};
+
+    if (ioctl(v->vm_fd, KVM_IRQFD, &irqfd) < 0)
+        throw_err("Failed to set the status of IRQFD");
+}
+
+void vm_ioeventfd_register(vm_t *v,
+                           int fd,
+                           unsigned long long addr,
+                           int len,
+                           int flags)
+{
+    struct kvm_ioeventfd ioeventfd = {
+        .fd = fd, .addr = addr, .len = len, .flags = flags};
```

Comment:
Add one comma (`,`) at the end of `flags`, so that `clang-format` can indent better.